scipy/signal/filter_design.py

```diff
@@ -362,37 +362,40 @@ def freqz(b, a=1, worN=None, whole=False, plot=None):
     except TypeError:  # not int-like
         w = atleast_1d(worN)
     else:
-        if worN <= 0:
+        if worN < 0:
             raise ValueError('worN must be positive, got %s' % (worN,))
         lastpoint = 2 * pi if whole else pi
         w = np.linspace(0, lastpoint, worN, endpoint=False)
-        min_size = b.size
-        if (a.size == 1 and worN >= min_size and
-                fftpack.next_fast_len(worN) == worN):
+        if (a.size == 1 and worN >= b.shape[0] and
+                fftpack.next_fast_len(worN) == worN and
+                (b.ndim == 1 or (b.shape[-1] == 1))):
             # if worN is fast, 2 * worN will be fast, too, so no need to check
             n_fft = worN if whole else worN * 2
             if np.isrealobj(b) and np.isrealobj(a):
                 fft_func = np.fft.rfft
             else:
                 fft_func = fftpack.fft
-            h = fft_func(b, n=n_fft)[:worN]
+            h = fft_func(b, n=n_fft, axis=0)[:worN]
             h /= a
             if fft_func is np.fft.rfft and whole:
                 # exclude DC and maybe Nyquist (no need to use axis_reverse
                 # here because we can build reversal with the truncation)
                 stop = -1 if n_fft % 2 == 1 else -2
                 h_flip = slice(stop, 0, -1)
                 h = np.concatenate((h, h[h_flip].conj()))
+            if b.ndim > 1:
+                # Last axis of h has length 1, so drop it.
+                h = h[..., 0]
+                # Rotate the first axis of h to the end.
+                h = np.rollaxis(h, 0, h.ndim)
     del worN
 
-    if w.ndim != 1:
-        raise ValueError('w must be 1D')
     if h is None:  # still need to compute using freqs w
-        if w.size == 0:
-            raise ValueError('w must have at least one element, got 0')
+        #if w.size == 0:
+        #    raise ValueError('w must have at least one element, got 0')
```

Comment:
delete?

Reply:
... and actually `worN=[]` and `worN=0` should probably be tested too

Reply:
Yup and yup.

Reply:
I just pushed an update.  There are now a few tests where `worN` is 0 or an array with size 0.